pkg/dataflatten/string_delimited.go

```diff
@@ -0,0 +1,84 @@
+package dataflatten
+
+import (
+	"bufio"
+	"bytes"
+	"strings"
+)
+
+type dataFunc func(data []byte, opts ...FlattenOpts) ([]Row, error)
+
+type recordSplittingStrategy int
+
+const (
+	None recordSplittingStrategy = iota + 1
```

Comment:
Why +1?

Reply:
Generally the advice is to have the uninitialize value (0) for an int, but the unknown/none case. So in this case, using iota makes a lot of sense. I think the other place in dataflatten that uses `iota + 1` there's no none case, so the 0 value is inherently an error. That might be confusing though

Reply:
Ok perfect! I was copying from elsewhere in dataflatten, but neglected to do the research to understand why the `+1` was used there. Thanks!